### PR TITLE
Add result index in ConstraintBasisStatus

### PR DESCRIPTION
**Pull request by blegat**
Closes https://github.com/JuliaOpt/MathOptInterface.jl/issues/11

### CODE REVIEW

File: src/attributes.jl [hunk 1]
```diff
@@ -776,13 +776,17 @@ ConstraintDual() = ConstraintDual(1)
...
+is 1 by default.
 
 **For the basis status of a variable, query the corresponding `SingleVariable`
 constraint that enforces the variable's bounds.**
 """
-struct ConstraintBasisStatus <: AbstractConstraintAttribute end
+struct ConstraintBasisStatus <: AbstractConstraintAttribute
+    N::Int
+end
+ConstraintDual() = ConstraintDual(1)
```
Comment (mlubin) [inline src/attributes.jl]: `ConstraintBasisStatus`?

**Comment by blegat** [comment]
> Are there any solvers that return multiple solutions with a basis for each? I'm not aware of any.

If they don't have the basis status for each solution, they should throw an error if the user get the basis status for a result that does not have the basis.

File: src/attributes.jl [hunk 2]
```diff
@@ -776,13 +776,17 @@ ConstraintDual() = ConstraintDual(1)
...
+A constraint attribute for the `BasisStatusCode` of some constraint in result
+`N`, with respect to an available optimal solution basis. If `N` is omitted, it
+is 1 by default.
 
 **For the basis status of a variable, query the corresponding `SingleVariable`
 constraint that enforces the variable's bounds.**
 """
-struct ConstraintBasisStatus <: AbstractConstraintAttribute end
+struct ConstraintBasisStatus <: AbstractConstraintAttribute
+    N::Int
```
Comment (mlubin) [inline src/attributes.jl]: This doesn't match the naming conventions for fields. What about `result_index`?

**Comment by mlubin** [comment]
My concern is that we're creating a new code path that will almost never be used and is likely to be incorrectly implemented by all solvers. Do we have a way to test that solvers don't ignore the result_index field?

File: src/attributes.jl (hunk 2 above)
Comment (blegat) [reply src/attributes.jl]: It's `N` for `VariablePrimal`, `ConstraintPrimal` and `ConstraintDual` but `result_index` for `ObjectiveValue` and `DualObjectiveValue`. `result_index` is indeed better.

**Comment by blegat** [comment]
This should be improved by https://github.com/JuliaOpt/MathOptInterface.jl/issues/900

File: src/attributes.jl [hunk 3]
```diff
@@ -776,13 +776,17 @@ ConstraintDual() = ConstraintDual(1)
 """
     ConstraintBasisStatus()
 
-A constraint attribute for the `BasisStatusCode` of some constraint, with
-respect to an available optimal solution basis.
+A constraint attribute for the `BasisStatusCode` of some constraint in result
+`N`, with respect to an available optimal solution basis. If `N` is omitted, it
```
Comment (mlubin) [inline src/attributes.jl]: s/`N`/`result_index`/

**Pull request merged**